safehttp/plugins/csp/csp.go

```diff
@@ -159,9 +162,30 @@ func (f FramingPolicy) Serialize(nonce string) string {
 	var b strings.Builder
 	b.WriteString("frame-ancestors 'self'")
 
-	if f.ReportURI != "" {
-		b.WriteString("; report-uri ")
-		b.WriteString(f.ReportURI)
+	ancestors := frameAncestors(f.Sources)
+	rep := report(f.ReportURI)
+	b.WriteString(strings.Join([]string{ancestors, rep}, "; "))
```

Comment:
I'm a bit nitpicky here, but please bear with me.

As it is, you have four lines:
1. Use `WriteString` to write a part of the `frame-ancestors` directive.
2. Create a string that contains a part of the `frame-ancestors` directive.
3. Create a string that contains the full `report-uri` directive.
4. Use `WriteString` to write the two parts above, joined by `"; "`.

Wouldn't it be cleaner to merge 1 and 2 inside `frameAncestors`?

This way the code would end up being:
```
var b strings.Builder
b.WriteString(frameAncestors(f.Sources)) // writes at least "frame-ancestors 'self'; " (note the space)
b.WriteString(report(f.ReportURI)) // writes a "; " delimited directive or an empty string
return strings.TrimSpace(b.String())
```

Reply:
That's true. It's simpler and avoids extra unneeded semicolon in the case report-uri is not present.